pkg/types/azure/validation/machinepool.go

```diff
@@ -81,6 +82,14 @@ func ValidateMachinePool(p *azure.MachinePool, poolName string, platform *azure.
 		}
 	}
 
+	if p.BootDiagnostics != nil {
```

Comment:
nit: Can you add an additional validation not to allow `Managed` when azure stack?